sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala

```diff
@@ -284,7 +285,10 @@ object RandomDataGenerator {
         new CalendarInterval(months, days, ns)
       })
       case _: DayTimeIntervalType => Some(() => Duration.of(rand.nextLong(), ChronoUnit.MICROS))
-      case _: YearMonthIntervalType => Some(() => Period.ofMonths(rand.nextInt()).normalized())
+      case YearMonthIntervalType(_, YEAR) =>
+        val period = Period.ofMonths(rand.nextInt()).normalized()
+        Some(() => Period.ofYears(period.getYears).normalized())
```

Comment:
can we just do `Period.ofYears(rand.nextInt() / 12).normalized())`?

Reply:
And I guess that `.normalized()` is not needed since we don't have other fields - so nothing to normalize.

Reply:
> And I guess that `.normalized()` is not needed since we don't have other fields - so nothing to normalize.

normalize is for checking equal   `P-1155869325M did not equal P-96322443Y-9M`

Reply:
> can we just do `Period.ofYears(rand.nextInt() / 12).normalized())`?

Updated

Reply:
did you push the commit?